include/vcmi/Creature.h

```diff
@@ -10,15 +10,29 @@
 
 #pragma once
 
-#include "Entity.h"
+#include "FactionMember.h"
 
 VCMI_LIB_NAMESPACE_BEGIN
 
 class CreatureID;
 class ResourceSet;
 enum class EGameResID : int8_t;
 
-class DLL_LINKAGE Creature : public EntityWithNativeTerrain<CreatureID>
+/// Base class for creatures and battle stacks
+class DLL_LINKAGE ACreature: public AFactionMember
+{
+public:
+	bool isLiving() const; //non-undead, non-non living or alive
```

Comment:
comment is complex, still don't get.
if it's about undead, suggest isUndead() with inverse usage.
Otherwise suggest isAlive()

Reply:
there are also gargoyles and elementals

Reply:
It is not about undead, but about any creatures that is neither:
- undead
- mechanical (e.g. War Machines)
- not "non-living", e.g. Golems
- and probably not a Gargoyle

In terms of game logic - this is a creature that is affected by spells like Death Ripple / Resurrection and such